File: Random/1.Top K Frequent Elements.js
/*
    Author : Mohamed Yousef 
    Date   : 2022-12-14
*/
let solve=function(nums,k){
    let defaultDict=new Proxy({},{
        get:(target,Name) => Name in target?target[Name]:0
    });
    nums.forEach(ele => {
        defaultDict[ele]+=1
    })
    // sort default dict
    //First create an array from Object 
    let item=Object.keys(defaultDict).map(function(key){
        return [key,defaultDict[key]]
    })
    item.sort(function(a,b){
        return b[1]-a[1]
    })
    // return item
    ans=[]
    for(let i=0;i<k;i++) {
        ans.push(parseInt(item[i][0]))
    }
    return ans
}
console.log(solve([1,1,1,2,2,2,2,3,3], 2));

File: Random/1.Top K Frequent Elements.py
#    Author : Mohamed Yousef 
#    Date   : 2022-12-14
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
sys.setrecursionlimit(50000) #To increase Recursion depth in py
def solve(nums , k):
    ourDict=defaultdict(int)
    for i in nums:
        ourDict[i] +=1
    ourDict=sorted(ourDict.items(),key=lambda ourDict:ourDict[1],reverse=True)
    # print(ourDict)
    ans=[]
    for i in range(k):
        ans.append(ourDict[i][0])
    return ans
print(solve([1,1,1,2,2,3], 2))
